### ISSUE TITLE
Templater: Add ability to attach functions to template slots

### ISSUE THREAD

**Issue by MynockSpit** [RecursiveThinking/recursive_thinking_website#41]
It's currently not possible to attach functions to dom elements with the templater. Figure out some pattern to attach functions to elements in templates.

Current Behaviour Example:

```js
// example function
function myFunc() {
  return 'string'
}

fill(templates.***my.template***, {
  attributeSlot: {
    onClick: "myFunc()" // won't work with Babel; will work if myFunc is global (on the window object)
    onMouseDown: myFunc() // won't work; function runs immediately
    onTouchStart: myFunc // won't work; attaches the body of the function as a string
  }
})
```

Rendered HTML:
```html
<element
  onclick = "myFunc()"
  onmousedown = "string"
  ontouchstart = "function myFunc() {
  return 'string'
}"
></element>
```

**Comment by amnevins** [comment]
I thought about this more and it does open an attack vector for stored xss.  If we arnt 10000 million percent confident in whatconing back and going to db this makes it easy to essentially eval() it.  Think about sliding a picture into a src attribute that is actually js xss attack....

**Comment by MynockSpit** [comment]
Nothing that's specified as an `onEVENT` attribute should be from the server. If it's not from the server, it's not an attack vector. What am I missing?

(It's important to note that currently, the templater trusts that everything passed into it is safe, so we're more as likely to get XSS from people's bios.)

**Comment by amnevins** [comment]
Yeah agreed there has to be a place somewhere in the app we leak for it to become reflective, however, the idea of templater is to map through some data and place that in the DOM.  We just have to be careful as we move toward this pattern that we are doing our due diligence to validate data going INTO the templater since its basically capable of eval() ing any code hat goes in.

**Comment by MynockSpit** [comment]
Yup! Agreed.

**Comment by MynockSpit** [comment]
@amnevins -- Do you have objections to this given that adding this capability doesn't make the templater any less safe (it's already unsafe).

**Comment by MynockSpit** [comment]
Wait, no. This isn't an attack vector. At no point are we going to be `eval`ing anything or appending it to the DOM, and everything you get from the server is going to be a string.
